ui/src/app/views/dataset/component.tsx

```diff
@@ -128,39 +137,57 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
     const {workspaceNamespace, workspaceFirecloudName} = resource;
     const {id} = updatedResource;
     let call;
+    let updatedConceptSetList = this.state.conceptSetList;
+    let updatedCohortList = this.state.cohortList;
     if (resource.cohort) {
       call = cohortsApi().updateCohort(workspaceNamespace, workspaceFirecloudName,
         id, updatedResource as Cohort);
+      updatedCohortList = this.state.cohortList.map((cohort) => {
+        return updatedResource.id === cohort.id ? (updatedResource as Cohort) : cohort;
+      });
     } else if (resource.conceptSet) {
       call = conceptSetsApi().updateConceptSet(workspaceNamespace, workspaceFirecloudName,
         id, updatedResource);
+      updatedConceptSetList = this.state.conceptSetList.map((conceptSet) => {
+        return updatedResource.id === conceptSet.id ? (updatedResource as ConceptSet) : conceptSet;
+      });
```

Comment:
It seems like there are a couple places with very similar code paths that only differ by a few pieces of information. It might be worth trying to refactor those at some point.